src/include/sof/audio/codec_adapter/codec_adapter.h

```diff
@@ -0,0 +1,13 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2020 Intel Corporation. All rights reserved.
+ *
+ * Author: Marcin Rajwa <marcin.rajwa@linux.intel.com>
+ */
+
+#ifndef __SOF_AUDIO_CODEC_ADAPTER__
+#define __SOF_AUDIO_CODEC_ADAPTER__
+
+#include <sof/audio/codec_adapter/codec/generic.h>
```

Comment:
What' the header used for  - seems to be some indirection only ?

Reply:
This header is used by `codec_adapter.h`, `generic.c` and cadence.c it provides "generic" data structures and interfaces shared beetween codec_adapter SoF component and specific codec like CADENCE ones

Reply:
ok, but I need to see the big picture. i.e. squash all commit for this new file and the others.